src/integrationTest/java/org/opensearch/test/framework/cluster/CloseableHttpClientFactory.java

```diff
@@ -50,6 +57,9 @@ public CloseableHttpClient getHTTPClient() {
 			.setDefaultSocketConfig(SocketConfig.custom().setSoTimeout(60, TimeUnit.SECONDS).build())
 			.build();
 		hcb.setConnectionManager(cm);
+		if(routePlanner != null) {
+			hcb.setRoutePlanner(routePlanner);
+		}
```

Comment:
out of curiosity, why do we need routePlanner?

Reply:
In order to set the request source IP address, this is done by `org.opensearch.test.framework.cluster.LocalAddressRoutePlanner` which implements `HttpRoutePlanner`. 

The Apache HTTP Client in version 4 contained a convenient method to set the request source IP address `RequestConfig.custom().setLocalAddress(inetAddress).build();`. But the method was removed in version 5 of Apache HTTP Client